File: src/main/java/axirassa/util/AutoSerializingObject.java

package axirassa.util;

import java.io.ByteArrayOutputStream;
import java.io.IOException;
import java.io.ObjectOutput;
import java.io.ObjectOutputStream;
import java.io.Serializable;

abstract public class AutoSerializingObject implements Serializable {
	private static final long serialVersionUID = -6381035456256744960L;


	public byte[] toBytes() throws IOException {
		ByteArrayOutputStream baos = new ByteArrayOutputStream();
		ObjectOutput out = new ObjectOutputStream(baos);
		out.writeObject(this);
		out.close();

		return baos.toByteArray();
	}
}

File: src/main/scala/axirassa/util/AutoSerializingObject.scala
package axirassa.util

import java.io.ByteArrayOutputStream
import java.io.ObjectOutputStream
import java.io.ByteArrayInputStream
import java.io.ObjectInputStream

trait AutoSerializingObject {
	def toBytes() = {
		val baos = new ByteArrayOutputStream
		val out = new ObjectOutputStream(baos)
		
		out.writeObject(this)
		out.close
		
		baos.toByteArray
	}
}

object AutoSerializingObject {
	def fromBytes(bytes: Array[Byte]) = { 
		val bais = new ByteArrayInputStream(bytes)
		val in = new ObjectInputStream(bais)
		
		try {
			in.readObject
		} finally {
			in.close
		}
	}
}